feature/basic_layout

aringlot opened the pull request:
refactoring according to review summary
translate

Commit:
feature/basic_layout

refactoring according to review summary
translate

Commit:
feature/basic_layout

refactoring according to review summary
translate

gskachkov commented on app/app.config.js:
```diff
@@ -1,4 +1,4 @@
-function routeConfig($stateProvider) {
+function config($stateProvider, $translateProvider, $translatePartialLoaderProvider) {
```
I also can be arrow function, it is better to use `ngInject` in this case. You forger to add translatePartialLoaderProvider within .$inject

gskachkov commented on app/app.config.js:
```diff
@@ -8,7 +8,15 @@ function routeConfig($stateProvider) {
-routeConfig.$inject = ['$stateProvider'];
-export default routeConfig;
\ No newline at end of file
+config.$inject = ['$stateProvider', '$translateProvider'];
```
Different number of parameters in function and in inject

gskachkov commented on app/app.module.js:
```diff
@@ -1,17 +1,24 @@
-     }
- ]);
+angular
+    .module('registratorApp', ['ui.router','pascalprecht.translate', 'ngCookies', 'common', 'aboutUs', 'home'])
```
import can return name of the module
so use can do in following manner:
```
import uiRouter from 'angular-ui-router';
...
angular.module('registratorApp', [uiRouter ... ])

```

gskachkov commented on app/app.module.js:
```diff
@@ -1,17 +1,24 @@
+            'ngInject'
+            $state.go('main');
+
+            $rootScope.$on('$translatePartialLoaderStructureChanged', function () {
```
Can this be replaced by ()=>$translate.refresh() ?

gskachkov commented on app/modules/common/components/button.component.js:
```diff
@@ -1,12 +1,15 @@
-    controllerAs: 'buttonCtrl'
+    template: template,
+    controller: Controller,
+    controllerAs: 'VM'
```
please use lowercase for vm

Commit:
feature/basic_layout

refactoring according to review summary

Commit:
feature/basic_layout

refactoring according to review summary

aringlot commented:
points, that were mentioned after review, are mostly fixed except bower -> npm

aringlot committed:
feature/basic_tests (#2)

* feature/basic_tests

unit tests are created here
also minor refactoring is done for tab.service

* feature/basic_tests

unit tests are created here
also minor refactoring is done for tab.service

Commit:
Merge remote-tracking branch 'origin/feature/basic_layout' into feature/basic_layout

# Conflicts:
#	app/dist/bundle.js
#	app/modules/home/home.module.js
#	webpack.config.js

Commit:
remove bundle.js
